backend/modules/vector_db/milvus.py

```diff
@@ -0,0 +1,285 @@
+                    a more performant Milvus server on [Docker or Kubernetes](https://milvus.io/docs/quickstart.md).
+                    In this setup, please use the server address and port as your uri, e.g.`http://localhost:19530`.
+                    If you enable the authentication feature on Milvus,
+                    use "<your_username>:<your_password>" as the token, otherwise don't set the token.
+                    - If you use [Zilliz Cloud](https://zilliz.com/cloud), the fully managed cloud
+                    service for Milvus, adjust the `uri` and `token`, which correspond to the
+                    [Public Endpoint and API key](https://docs.zilliz.com/docs/on-zilliz-cloud-console#cluster-details)
+            -   api_key: str
+                Token for authentication with the Milvus server.
+        """
+        # TODO: create an extended config for Milvus like done in Qdrant
+        logger.debug(f"Connecting to Milvus using config: {config.model_dump()}")
+        if config.local is True:
+            self.milvus_client = MilvusClient(
+                "./cognita_milvus.db",
+                db_name=config.config.get("db_name", "milvus_default_db"),
+            )
+        else:
+            url = config.url
+            api_key = config.api_key
+            if not api_key:
+                api_key = None
+
+            self.milvus_client = MilvusClient(
+                uri=url,
+                token=api_key,
+                db_name=config.config.get("db_name", "milvus_default_db"),
+            )
+
+    def create_collection(self, collection_name: str, embeddings: Embeddings):
+        """
+        Create a collection in the vector database
+        Args:
+        :param collection_name: str - Name of the collection
+        :param embeddings: Embeddings - Embeddings object to be used for creating embeddings of the documents
+        Current implementation includes Quick setup in which the collection is created, indexed and loaded into the memory.
+
+        """
+        # TODO: Add customized setup with indexed params
+        logger.debug(f"[Milvus] Creating new collection {collection_name}")
+
+        vector_size = self.get_embedding_dimensions(embeddings)
+
+        fields = [
+            FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
+            FieldSchema(name="vector", dtype=DataType.FLOAT_VECTOR, dim=vector_size),
+            FieldSchema(name="text", dtype=DataType.VARCHAR, max_length=65535),
+            FieldSchema(name="metadata", dtype=DataType.JSON),
+        ]
+
+        schema = CollectionSchema(
+            fields=fields, description=f"Collection for {collection_name}"
+        )
+
+        self.milvus_client.create_collection(
+            collection_name=collection_name,
+            dimension=vector_size,
+            metric_type="COSINE",  # https://milvus.io/docs/metric.md#Metric-Types : check for other supported metrics
```

Comment:
Sure, by `make this configurable` do you mean `metric_type` ? 
The definition of best metric actually is subjective, it depends on the data and the use caase actually as per my understanding. if let's say the embedding vectors are sparse and our use case demands a full text exact search then `BM25` is more preferrable, if data is not normalized then `Inner Product (IP)` would make much more sense, but in general for dense vectors `cosine similarity` captures the orientation of the vectors and hence the semantic meaning and is default choice in lot of vector DBs.

Reply:
>Sure, by make this configurable do you mean metric_type ?

Yes, what do you think?

Reply:
Yes, my initial thought was to make it configurable and there are actually multiple paths to do it, since this is a generic / common field across all attributes so it makes more sense to me to add it directly under `VectorDBConfig` as an attribute but that would demand changes in rest of the vectordb implementations so to avoid that for now, I am thinking of leveraging the `config` attribute which is meant for additional params.

Will update the code in the next commit and pls lemme know if that implementation looks good to start with.